src/kronecker.jl

```diff
@@ -46,24 +44,30 @@ julia> Matrix(Δ)
   0   1   1  -4
 ```
 """
-Base.kron(A::LinearMap, B::LinearMap) = KroneckerMap{promote_type(eltype(A), eltype(B))}((A, B))
-Base.kron(A::LinearMap, B::KroneckerMap) = KroneckerMap{promote_type(eltype(A), eltype(B))}(tuple(A, B.maps...))
-Base.kron(A::KroneckerMap, B::LinearMap) = KroneckerMap{promote_type(eltype(A), eltype(B))}(tuple(A.maps..., B))
-Base.kron(A::KroneckerMap, B::KroneckerMap) = KroneckerMap{promote_type(eltype(A), eltype(B))}(tuple(A.maps..., B.maps...))
-Base.kron(A::LinearMap, B::LinearMap, Cs::LinearMap...) = KroneckerMap{promote_type(eltype(A), eltype(B), map(eltype, Cs)...)}(tuple(A, B, Cs...))
+Base.kron(A::LinearMap, B::LinearMap) =
+    KroneckerMap{promote_type(eltype(A), eltype(B))}((A, B))
+Base.kron(A::LinearMap, B::KroneckerMap) =
+    KroneckerMap{promote_type(eltype(A), eltype(B))}(tuple(A, B.maps...))
+Base.kron(A::KroneckerMap, B::LinearMap) =
+    KroneckerMap{promote_type(eltype(A), eltype(B))}(tuple(A.maps..., B))
+Base.kron(A::KroneckerMap, B::KroneckerMap) =
+    KroneckerMap{promote_type(eltype(A), eltype(B))}(tuple(A.maps..., B.maps...))
+Base.kron(A::LinearMap, B::LinearMap, Cs::LinearMap...) =
+    KroneckerMap{promote_type(eltype(A), eltype(B), map(eltype, Cs)...)}(tuple(A, B, Cs...))
+    # could this just be a recursive definition: kron(A, B, C) = kron(kron(A, B), C) ?
```

Comment:
One could try. I wanted to make sure in general that when you have first a couple of matrices and only eventually a `LinearMap`, then we don't want to `kron` the matrices first, but we want it all lazy. But I think we have a test for that. But my guess is that I had tried that already (in fact, it's the generic base behaviour).

Reply:
Oh right, I did not think about that. Although wait, this is a method for `LinearMap` objects only, right, so the recursive definition will just end up calling `kron(::LinearMap, ::LinearMap)` which create a `KroneckerMap`, no?

Reply:
True. I didn't remember because it wasn't shown in the diff, but making sure that, even when you have several matrices first, that those are lazily Kronecker multiplied, is the task of the "symbolic loop" below. So, in fact, this `A, B, Cs...` method could be removed, it seems.